controllers/storagecluster/cephcluster.go

```diff
@@ -167,10 +167,11 @@ func (obj *ocsCephCluster) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.
 		cephCluster = newExternalCephCluster(sc, monitoringIP, monitoringPort)
 	} else {
 		// Add KMS details to CephCluster spec, only if
-		// cluster-wide encryption is enabled
-		// ie, sc.Spec.Encryption.ClusterWide/sc.Spec.Encryption.Enable is True
+		// cluster-wide encryption is enabled or any of the device set is encrypted
+		// ie, sc.Spec.Encryption.ClusterWide/sc.Spec.Encryption.Enable is True or any device is encrypted
 		// and KMS ConfigMap is available
-		if sc.Spec.Encryption.Enable || sc.Spec.Encryption.ClusterWide {
+
+		if sc.Spec.Encryption.Enable || sc.Spec.Encryption.ClusterWide || util.IsAnyDeviceSetEncrypted(sc.Spec.StorageDeviceSets) {
```

Comment:
This check for multiple conditions is repeated in several places. Could this be refactored to a helper method?

Reply:
Done